sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +383,512 @@ def transform(self, X):
+        Each tuple has `(feat_idx, neighbor_feat_idx, predictor)`, where
+        `feat_idx` is the current feature to be imputed, `neighbor_feat_idx`
+        is the list of other features used to impute the current feature, and
+        `predictor` is the trained predictor used for the imputation.
+
+    Notes
+    -----
+    Features which contain only missing values at ``fit`` are discarded upon
+    ``transform``.
+
+    Features with missing values in transform which did not have any missing
+    values in fit will be imputed with the initial imputation method.
+
+    References
+    ----------
+    .. [1] `Stef van Buuren, Karin Groothuis-Oudshoorn (2011). "mice:
+        Multivariate Imputation by Chained Equations in R". Journal of
+        Statistical Software 45: 1-67.
+        <https://www.jstatsoft.org/article/view/v045i03>`_
+    """
+
+    def __init__(
+            self,
+            missing_values='NaN',
+            imputation_order='monotone',
+            n_imputations=100,
+            n_burn_in=10,
+            predictor=None,
+            n_nearest_features=None,
+            initial_strategy="mean",
+            min_value=None,
+            max_value=None,
+            verbose=False,
+            random_state=None):
+
+        self.missing_values = missing_values
+        self.imputation_order = imputation_order
+        self.n_imputations = n_imputations
+        self.n_burn_in = n_burn_in
+        self.predictor = predictor
+        self.n_nearest_features = n_nearest_features
+        self.initial_strategy = initial_strategy
+        self.min_value = min_value
+        self.max_value = max_value
+        self.verbose = verbose
+        self.random_state = random_state
+
+    def _impute_one_feature(self,
+                            X_filled,
+                            mask_missing_values,
+                            feat_idx,
+                            neighbor_feat_idx,
+                            predictor=None):
+        """Imputes a single feature from the others provided.
+
+        This function predicts the missing values of one of the features using
+        the current estimates of all the other features. The `predictor` must
+        support `return_std=True` in its ``predict`` method for this function
+        to work.
+
+        Parameters
+        ----------
+        X_filled : array-like
+            Input data with the most recent imputations.
+
+        mask_missing_values : array-like
+            Input data's missing indicator matrix.
+
+        feat_idx : integer
+            Index of the feature currently being imputed.
+
+        neighbor_feat_idx : array-like
+            Indices of the features to be used in imputing `feat_idx`.
+
+        predictor : object, default=BayesianRidgeRegression
+            The predictor to use at this step of the round-robin imputation.
+            It must support ``return_std`` in its ``predict`` method.
+            If None, it will be fit, otherwise not.
```

Comment:
This is confusing. If none, surely it can't be fit

Reply:
Sorry, I changed it to: 
            The predictor to use at this step of the round-robin imputation.
            It must support ``return_std`` in its ``predict`` method.
            If None, it will be cloned from self._predictor and fit. Otherwise
            if a predictor is provided, it will not be fit but used only to
            predict.